src/ProjectOrigin.WalletSystem.Server/Activities/TransferFullSliceActivity.cs

```diff
@@ -43,6 +45,7 @@ public async Task<ExecutionResult> Execute(ExecuteContext<TransferFullSliceArgum
         try
         {
             var sourceSlice = await _unitOfWork.CertificateRepository.GetWalletSlice(context.Arguments.SourceSliceId);
+            var sourceEndpoint = await _unitOfWork.WalletRepository.GetWalletEndpoint(sourceSlice.WalletEndpointId);
```

Comment:
It seems a bit annoying that we need to fetch the wallet endpoint here in order to get the WalletId.

Reply:
Indeed, speaks to the point we have had earlier that it might make sense to later rework how some of these objects work

Reply:
agree